util/lru_cache.cc

```diff
@@ -259,14 +259,11 @@ Cache::Handle* LRUCacheShard::Lookup(const Slice& key, uint32_t hash) {
 bool LRUCacheShard::Ref(Cache::Handle* h) {
   LRUHandle* handle = reinterpret_cast<LRUHandle*>(h);
   MutexLock l(&mutex_);
-  if (handle->InCache()) {
-    if (handle->refs == 1) {
-      LRU_Remove(handle);
-    }
-    handle->refs++;
-    return true;
+  if (handle->InCache() && handle->refs == 1) {
+    LRU_Remove(handle);
   }
-  return false;
+  handle->refs++;
```

Comment:
Yeah agree both are correct. We can avoid extra lookups if Ref() succeeds in case where handle is externally-referenced-only (if it fails the caller would have to do an unnecessary Lookup()). Let me clarify the description.